sqlglot/dataframe/sql/column.py

```diff
@@ -0,0 +1,330 @@
+from __future__ import annotations
+
+import datetime
+import typing as t
+from collections.abc import Iterable
+
+import sqlglot
+from sqlglot import expressions as exp
+from sqlglot.dataframe.sql.types import DataType
+from sqlglot.helper import flatten
+
+if t.TYPE_CHECKING:
+    from sqlglot.dataframe.sql._typing import (
+        ColumnOrName,
+        ColumnOrPrimitive,
+        DateTimeLiteral,
+        DecimalLiteral,
+        Literals,
+        Primitives,
+    )
+    from sqlglot.dataframe.sql.window import WindowSpec
+
+
+class Column:
+    def __init__(self, expression: t.Union[t.Any, exp.Expression]):
```

Comment:
[Nitpick] Seems like we expect very specific types in the constructor. Should we enumerate them in the Union instead of just using Any?

Reply:
Yes this was a bad shortcut that I forgot to fix. Will address this as I integrate mypy and clean up the existing types.